django_mongodb/compiler.py

```diff
@@ -91,16 +81,7 @@ def _make_result(self, entity, columns, converters, tuple_expected=False):
                 if column_alias is not None and column_alias != self.collection_name
                 else entity
             )
-            value = obj.get(name, NOT_PROVIDED)
-            if value is NOT_PROVIDED:
-                value = field.get_default()
-            elif converters:
-                # Decode values using Django's database converters API.
-                for converter in converters.get(name, ()):
-                    value = converter(value, col, self.connection)
-            result.append(value)
-        if tuple_expected:
-            result = tuple(result)
+            result.append(obj.get(name, col.field.get_default()))
```

Comment:
I see you added the `get_default()` bit back in a separate commit. Were there any test failures before that? The code came from the mongo engine fork and I've been skeptical of it. It seems it could be related to missing columns due to schema changes, but I'm not sure it's the behavior we want. Just wondering if you have any analysis to add.

Reply:
Yes, there was. A JSONField test, don't remember the name (not_provided wasn't a json). But basically I copied the logic from Django. It is not that deep like types changes or so. 
Currently we are flagging those fields with NOT_PROVIDED, then we check if it not provided we assign the default, other case we apply converters. you can see the code here:
https://github.com/WaVEV/django-mongodb/blob/main/django_mongodb/compiler.py#L94
Now a default value is assigned and then don't care about the converters. The only thing to point out is: the code is running converters over default values. But Django does the same if I am not mistaken.